editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -27,29 +27,50 @@ const ALLOW_LIST = [
   "drawBoundary.action",
 ] as const;
 
-export type HelpClickMetadata = Record<string, string>;
-export type SelectedUrlsMetadata = Record<"selectedUrls", string[]>;
-export type BackwardsNavigationInitiatorType = "change" | "back";
+type HelpClickMetadata = Record<"hasClickedHelp", boolean>;
+type SelectedUrlsMetadata = Record<"selectedUrls", string[]>;
+type BackwardsNavigationInitiatorType = "change" | "back";
 
-type NodeMetadata = {
+type ResultNodeMetadata = {
+  flag?: Flag;
   flagSet?: FlagSet;
   displayText?: {
     heading?: string;
     description?: string;
   };
-  flag?: Flag;
-  title?: string;
-  type?: string | null;
-  id?: string;
+};
+
+type AutoAnswerMetadata = {
   isAutoAnswered?: boolean;
 };
 
+type NodeMetadata = ResultNodeMetadata & AutoAnswerMetadata;
+
+type BackwardsTargetMetadata = {
+  id: string;
+  type: string | null;
+  title: string;
+};
+
+type BackwardsNavigationMetadata =
+  | Record<"change", BackwardsTargetMetadata>
+  | Record<"back", BackwardsTargetMetadata>;
+
+/**
+ * Describes the value held in analytics_logs.metadata
+ */
+export type Metadata =
+  | NodeMetadata
+  | BackwardsNavigationMetadata
+  | SelectedUrlsMetadata
+  | HelpClickMetadata;
+
 let lastVisibleNodeAnalyticsLogId: number | undefined = undefined;
 
 const analyticsContext = createContext<{
   createAnalytics: (type: AnalyticsType) => Promise<void>;
-  trackHelpClick: (metadata?: HelpClickMetadata) => Promise<void>;
-  trackNextStepsLinkClick: (metadata?: SelectedUrlsMetadata) => Promise<void>;
+  trackHelpClick: (metadata?: Metadata) => Promise<void>;
+  trackNextStepsLinkClick: (metadata?: Metadata) => Promise<void>;
```

Comment:
It seems that this change means the type is _less_ desscriptive/meaningful rather than _more_ - I'm not sure I understand? 

If `trackHelpClick` has type `HelpClickMetadata` then it already _is_ valid `Metadata` right? But with this update, I could incorrectly pass `trackHelpClick({ "back": { ...metadata } })` and I don't think it wouldn't get caught, right?

Reply:
If I understand Daf's intention with adding `Metadata` I think it's nice to make clear the expected content that could be written to the `metadata` jsonb column which _could_ have anything written to it but we're setting our expectations which I really like and could see the value of. I was associating all functions which write to the column with `Metadata` so that it's clear it'll be one of the types it's composed of.

Although, as you say it does mean we're not enforcing the types on the functions which as you say is less descriptive. 

I've updated my PR to expand on the `Metadata` but the new type never gets used anywhere preserving the more explicit types on the functions 🤔 

Rather than go with this approach it would maybe be best to remove the `Metadata` type?

Reply:
In my mind this distinction is: 
- specific types like `HelpClickMetadata` should be used to restrict the arguments passed to methods like `trackHelpClick` 
- the broad union type `Metadata` should be used to define what the GraphQL mutation/query requests for this JSONB column expect

Reply:
I definitely agree with that @jessicamcinchak although the bit I'm not sure on is whether we can actually assign the `Metadata` type to the GraphQL request, is that what you're suggesting? 

With `trackHelpClick` I think the function is associated with `HelpClickMetadata` but there's no where to add `Metadata` as I understand it although I think I might be missing something.

```
  async function trackHelpClick(metadata?: HelpClickMetadata) {
    if (shouldSkipTracking()) return;

    await publicClient.mutate({
      mutation: gql`
        mutation UpdateHasClickedHelp($id: bigint!, $metadata: jsonb = {}) {
          update_analytics_logs_by_pk(
            pk_columns: { id: $id }
            _set: { has_clicked_help: true }
            _append: { metadata: $metadata }
          ) {
            id
          }
        }
      `,
      variables: {
        id: lastVisibleNodeAnalyticsLogId,
        metadata,
      },
    });
  }
  ```
  
Or would you suggest we maintain the `Metadata` but just have it in the file not being called but serving as a marker?

I'm a little bamboozled I think but it would be good to fix this PR / close it as I fear I'm wasting your time with this.